File: js/223_RectangleArea.js
/**
 * @param {number} A
 * @param {number} B
 * @param {number} C
 * @param {number} D
 * @param {number} E
 * @param {number} F
 * @param {number} G
 * @param {number} H
 * @return {number}
 */
var computeArea = function(A, B, C, D, E, F, G, H) {
    var left = Math.max(A,E);
    var right = Math.max( left, Math.min(C,G));
    var bottom = Math.max(B,F);
    var top = Math.max(bottom, Math.min(D,H));
    return (C-A) * (D-B) + (G-E) * (H-F) - (right-left) * (top-bottom);
};

File: cpp/223_RectangleArea.cpp
class Solution {
public:
    int computeArea(int A, int B, int C, int D, int E, int F, int G, int H) {
        int left = std::max(A,E);
        int right = std::max(left, std::min(C,G));
        int bottom = std::max(B,F);
        int top = std::max(bottom, std::min(D,H));
        return (C-A) * (D-B) + (G-E) * (H-F) - (right - left) * (top - bottom);
    }
    
};